### ISSUE TITLE
crash - not enough memory (catching up replay)

### ISSUE THREAD

**Issue by mojjj** [ZeroK-RTS/Zero-K#3511]
linux 64, ubuntu 16.04, steam, 32G ram, running command

cd ~/.local/share/Steam/steamapps/common/Zero-K && ./Zero-K_linux64.sh

#####
infolog
https://pastebin.com/5hFq8SPk

**Comment by mojjj** [comment]
and this is not the first time

**Comment by sprunk** [comment]
See the comment at https://github.com/spring/spring/blob/3d159a0e675c350f11598b9a3a6d2412f3d00af5/rts/Game/Game.cpp#L1529

We willingly brought this on ourselves: ec385c2e936257c9b80d5e3a49da3bd81e5b0d34

**Comment by Anarchid** [comment]
That will go back to very slow (including less than second per second) catch up speeds. Make it configurable in some way perhaps?

**Comment by sprunk** [comment]
Yes, it will necessarily have to be slower because it has to spend time cleaning up all the unused memory that otherwise causes an OOM. But sure, let's make a checkbox:

 - [x] crash randomly when catching up
